File: app/src/main/java/ai/elimu/content_provider/room/dao/StoryBookParagraph_WordDao.java
package ai.elimu.content_provider.room.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import ai.elimu.content_provider.room.entity.StoryBookParagraph_Word;

@Dao
public interface StoryBookParagraph_WordDao {

    @Insert
    void insert(StoryBookParagraph_Word storyBookParagraph_Word);

    @Update
    void update(StoryBookParagraph_Word storyBookParagraph_Word);

    @Query("DELETE FROM StoryBookParagraph_Word WHERE StoryBookParagraph_id = :storyBookParagraphId")
    void delete(Long storyBookParagraphId);

    @Query("DELETE FROM StoryBookParagraph_Word")
    void deleteAll();
}

File: app/src/main/java/ai/elimu/content_provider/room/dao/StoryBookParagraph_WordDao.kt
package ai.elimu.content_provider.room.dao

import ai.elimu.content_provider.room.entity.StoryBookParagraph_Word
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

@Dao
interface StoryBookParagraph_WordDao {
    @Insert
    fun insert(storyBookParagraph_Word: StoryBookParagraph_Word)

    @Update
    fun update(storyBookParagraph_Word: StoryBookParagraph_Word)

    @Query("DELETE FROM StoryBookParagraph_Word WHERE StoryBookParagraph_id = :storyBookParagraphId")
    fun delete(storyBookParagraphId: Long?)

    @Query("DELETE FROM StoryBookParagraph_Word")
    fun deleteAll()
}
